src/main/java/qtlog/LogParser/LogParser.java

```diff
@@ -0,0 +1,257 @@
+package qtlog.LogParser;
+
+import qtlog.shared.FileDTO;
+import qtlog.shared.RawLogDTO;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class LogParser implements ILogParser{
+    private int agentCount;
+    private int skillCount;
+    @Override
+    public RawLogDTO giveLogToParse(FileDTO fileInfo) {
```

Comment:
This function name doesn't describe what the function does... Instead it describes how you use the parameters??? I'm so confused. 
Shouldn't this be called `readLog`? or something similar?